### ISSUE TITLE
Simplifying the difference of two sympy.Eq returns -1

### ISSUE THREAD

**Issue by leerobert** [sympy/sympy#10422]
Unless the two Equality expressions are identical, the difference of two equality expressions simplified returns -1. Is this a limitation of the simplify function or is there another way to handle linear systems operations? 

```
>>> expr = sympy.Eq(2*x,5)
>>> expr2 = sympy.Eq(2*x,5)
>>> sympy.simplify(expr - expr2)
0

>>> expr = sympy.Eq(2*x-4,7)
>>> expr2 = sympy.Eq(2*x,11)
>>> sympy.simplify(expr - expr2)
-1
```

I see the result of subtracting two expressions is:

```
>>> expr - expr2
-(2*x == 11) + (2*x - 4 == 7)
```

Does this require handling each part of the equality separately when handling systems of equations?

**Comment by MSeeker** [comment]
The current development version of SymPy (0.7.7.dev) no longer simplifies an `Eq` to `-1`. For example (This is the python 3.5 version, the 2.7 version works the same):
```python
In [1]: from sympy import *

In [2]: x = symbols('x')

In [3]: eq1 = Eq(2 * x - 4, 7); eq2 = Eq(2 * x, 11)

In [4]: eq1 - eq2
Out[4]: -(Eq(2*x, 11)) + (Eq(2*x - 4, 7))

In [5]: simplify(_)
Out[5]: -(Eq(2*x, 11)) + (Eq(2*x - 4, 7))
```
Nevertheless, I fail to see why arithmetic operation between `Eq`s are not supported in SymPy. After all, "adding X multiples of eq.1 to eq.2" should be one of the most elementary operations performed on equations. This feature should be easy to implement though (makes a good project for my first commit ^^), I'll see if I could come up with a solution.

**Comment by MSeeker** [comment]
Did a little digging and found the exact issue was already dressed in #8023. So the general idea is that `Eq` (as well as all subclasses of `Relational`) should behave like booleans in the context of arithmetic operations. For example, `Eq(1, 1) + Eq(2, 2)` should return `2` (being a result of `True` + `True`) instead of a single `True` from `Eq(3, 3)`.
This does make equation manipulation a bit inconvenient, but I think we can always get around it by moving all rhs to the left and use the more mature `Expr` manipulation functionalities.

**Comment by leerobert** [comment]
I figured -1 being returned during simplification. was probably the result of a comparison on boolean operations (since subtracting the two equations returned a convoluted boolean expression). I was going to go ahead and write some logic that does add, sub, mult, and div operations on expressions as that is my current use case. Whatever I write up, I can contribute here. The simplify function was a little too much for me to digest in one sitting so nice work in digging deeper into the matter... your google skills are clearly on point ;).

**Comment by micklat** [comment]
I've written a small Eqn class for my own purposes and found it very useful. I'm new to sympy so maybe I'm missing a better way to do this. I'm afraid I'm not very interested in working this into a pull request, but if somebody else would like to pick it up and add it so sympy, be my guest.

import sympy
from sympy import S
import operator

class Eqn(object):
  """
  This class represents an equation. It is meant for reasoning with equations: adding
  equations, subtracting equations, multiplying them, making substitutions etc.

  >>> from sympy import var
  >>> var('x')
  x
  >>> eq1 = Eqn(2*x, 5)
  >>> eq1 += 4
  >>> eq1.solve()
  {x: 5/2}
  """

  __slots__ = ['sides']

  def __init__(self, *sides):
    self.sides = map(S, sides)
  
  def map(self, *fs):
    res = self
    for f in fs:
      res = Eqn(*map(f, res.sides))
    return res

  def __iadd__(self, other):
    self.sides = (self+other).sides
    return self

  def __imul__(self, other):
    self.sides = (self*other).sides
    return self

  def __isub__(self, other):
    self.sides = (self-other).sides
    return self

  def __itruediv__(self, other):
    self.sides = operator.truediv(self,other).sides
    return self
  
  def __rmul__(self, other):
    return self.binary_op(operator.mul, other, self)

  def __radd__(self, other):
    return self.binary_op(operator.add, other, self)
    
  def __rtruediv__(self, other):
    return self.binary_op(operator.truediv, other, self)

  def __rsub__(self, other):
    return self.binary_op(operator.sub, other, self)

  def __neg__(self):
    return self.map(operator.neg)

  def as_eqs(self):
    return [sympy.Eq(self.sides[i-1], self.sides[i])
            for i in xrange(1, len(self.sides))]
  
  def solve(self, *args, **kwArgs):
    return sympy.solve(self.as_eqs(), *args, **kwArgs)

  def isolate(self, x, lhs=0, rhs=1):
    solution = sympy.solve(self.as_eqs(), x, dict=True)
    if isinstance(solution,list):
      solution, = solution
    return Eqn(x, solution[x])
  
  def __str__(self):
    return ' = '.join(map(str,self.sides))

  def foreach(self, *fs):
    for f in fs:
      self.sides = map(f, self.sides)

  def __getitem__(self,idx):
    return self.sides[idx]

  def __setitem__(self,idx,val):
    self.sides[idx] = val

  @classmethod
  def binary_op(cls, op, a, b):
    if isinstance(b, Eqn) and isinstance(a, Eqn):
      return Eqn(*map(op, a.sides, b.sides))
    if isinstance(a, Eqn):
      return Eqn(*[op(x,b) for x in a.sides])
    assert isinstance(b, Eqn)
    return Eqn(*[op(a,y) for y in b.sides])
  
  def __pow__(self, other):
    return self.binary_op(operator.pow, self, other)
      
  def __mul__(self, other):
    return self.binary_op(operator.mul, self, other)

  def __truediv__(self, other):
    return self.binary_op(operator.truediv, self, other)

  def __add__(self, other):
    return self.binary_op(operator.add, self, other)

  def __sub__(self, other):
    return self.binary_op(operator.sub, self, other)

  def reverse(self):
    return Eqn(*self.sides[::-1])

  def revert(self):
    self.sides = self.sides[::-1]

  def sides_ratio(self, numerator=1, denominator=0):
    return operator.truediv(self.sides[numerator], self.sides[denominator])

  def sides_diff(self, added=1, subtracted=0):
    return self.sides[added]-self.sides[subtracted]

  def subs(self, other):
    return self.map(lambda s: s.subs(other.sides[0], other.sides[1]))

if __name__=='__main__':
  import doctest
  doctest.testmod()

**Comment by oscarbenjamin** [comment]
The simplification giving -1 problem is solved on current master but arithmetic operations on `Relational` are still nonsensical.

I have a PR #18053 which disables arithmetic operations on `Relational` by making it not subclass `Expr`. That solves the first problem in this issue:
```julia
In [1]: simplify(Eq(2*x-4, 7) - Eq(2*x, 11))                                                                                      
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-0f816a89db9e> in <module>
----> 1 simplify(Eq(2*x-4, 7) - Eq(2*x, 11))

TypeError: unsupported operand type(s) for -: 'Equality' and 'Equality'
```

What remains then is introducing a new `Equation` (or something) class that actually implements arithmetic in the obvious way that is wanted. That is discussed in #8023 but the issue for it is #4986.

I propose that if #18053 is merged then this issue can be closed in favour of #4986 to introduce a new `Equation` class.
